aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu

```diff
@@ -120,12 +120,39 @@ void i1e_kernel_cuda(TensorIteratorBase& iter) {
   #endif
 }
 
+const char sigmoid_name[] = "sigmoid";
 void sigmoid_kernel_cuda(TensorIteratorBase& iter) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.common_dtype(), "sigmoid_cuda", [&]() {
-    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
-      return static_cast<scalar_t>(1) / (static_cast<scalar_t>(1) + std::exp(-a));
+  auto common_dtype = iter.common_dtype();
+  if (at::isComplexType(common_dtype)) {
+    // only jiterate for complex-dtype
+    #if AT_USE_JITERATOR()
+      static const auto sigmoid_string = jiterator_stringify(
```

Comment:
should we add this in `aten/src/ATen/native/cuda/Math.cuh` along with the other op strings?

Reply:
The reason for that was `sigmoid` is a relatively trivial op that does not depend on other operator strings from `Math.cuh`. Also touching `Math.cuh` leads to building of multiple files so kept it here.